x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/openai/OpenAiEmbeddingsResponseEntity.java

```diff
@@ -0,0 +1,205 @@
+                return;
+            }
+        }
+
+        throw new IllegalStateException(format("Failed to find required field [%s] in OpenAI embeddings response", field));
+    }
+
+    private static Embedding parseEmbeddingObject(XContentParser parser) throws IOException {
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+
+        positionParserAtTokenAfterField(parser, "embedding");
+
+        List<Float> embeddingValues = XContentParserUtils.parseList(parser, OpenAiEmbeddingsResponseEntity::parseEmbeddingList);
+
+        // the parser is currently sitting at an ARRAY_END so go to the next token
+        parser.nextToken();
+        // if there are additional fields within this object, lets skip them, so we can begin parsing the next embedding array
+        parser.skipChildren();
+
+        return new Embedding(embeddingValues);
+    }
+
+    private static float parseEmbeddingList(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.currentToken();
+        XContentParserUtils.ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token, parser);
+        return parser.floatValue();
+    }
+
+    public record Embedding(List<Float> values) implements Writeable, ToXContentObject {
+        public static final String EMBEDDING = "embedding";
+
+        public Embedding(StreamInput in) throws IOException {
+            this(in.readCollectionAsImmutableList(StreamInput::readFloat));
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            out.writeCollection(values, StreamOutput::writeFloat);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            builder.field(EMBEDDING, values);
+            builder.endObject();
+            return builder;
+        }
+
+        @Override
+        public String toString() {
+            return Strings.toString(this);
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field(TEXT_EMBEDDING, embeddings);
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeCollection(embeddings);
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public String getResultsField() {
+        return TEXT_EMBEDDING;
+    }
+
+    @Override
+    public Map<String, Object> asMap() {
+        Map<String, Object> map = new LinkedHashMap<>();
+        map.put(getResultsField(), embeddings);
+
+        return map;
+    }
+
+    @Override
+    public Map<String, Object> asMap(String outputField) {
+        Map<String, Object> map = new LinkedHashMap<>();
+        map.put(outputField, embeddings);
```

Comment:
Let me know if this looks right.

Reply:
This was wrong 😆 . When I tested with the ingest processor it was failing so it needs to look more like this
```
map.put(outputField, embeddings.stream().map(Embedding::asMap).collect(Collectors.toList()));
```